src/Authentication/MsalManagedIdentityTokenProvider.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.AppConfig;
+
+namespace Microsoft.Artifacts.Authentication;
+
+public class MsalManagedIdentityTokenProvider : ITokenProvider
+{
+    private readonly ILogger logger;
+    private readonly IAppConfig appConfig;
+
+    public MsalManagedIdentityTokenProvider(IAppConfig appConfig, ILogger logger)
+    {
+        this.appConfig = appConfig;
+        this.logger = logger;
+    }
+
+    public string Name => "MSAL Managed Identity";
+
+    public bool IsInteractive => false;
+
+    public bool CanGetToken(TokenRequest tokenRequest) => 
```

Comment:
we should check against system here, so if the env var is not set or not system we don't try, otherwise we'll try